src/js/components/Factoid.js

```diff
@@ -0,0 +1,49 @@
+// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP
+
+import React, { Component, PropTypes } from 'react';
+import Paragraph from './Paragraph';
+
+const CLASS_ROOT = 'factoid';
+
+export default class Factoid extends Component {
+  render() {
+    let classes = [CLASS_ROOT];
+    let IconBefore = undefined;
+    let IconAfter = undefined;
+
+    if (this.props.size) {
+      classes.push(`${CLASS_ROOT}--${this.props.size}`);
+    }
+    if (this.props.className) {
+      classes.push(this.props.className);
+    }
+
+    if (this.props.iconBefore) {
+      const Icon = require('grommet/components/icons/base/' + this.props.iconBefore);
```

Comment:
I would rather have the user send the icon instance. Dynamic require is problematic with Node. In this case, webpack will include all the icons in the bundle because it does not know which one will be passed along.

So we need to convert `iconBefore: PropTypes.string,` to `iconBefore: PropTypes.node,`